ingestion-sink/src/test/java/com/mozilla/telemetry/ingestion/sink/io/PubsubReadIntegrationTest.java

```diff
@@ -12,22 +12,22 @@
 import java.util.LinkedList;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 import org.junit.Test;
 
 public class PubsubReadIntegrationTest extends TestWithPubsubResources {
 
+  private static final PubsubMessage TEST_MESSAGE = PubsubMessage.newBuilder()
+      .setData(ByteString.copyFrom("test".getBytes())).build();
+
   @Test
-  public void canReadOneMessage() throws Exception {
-    publisher
-        .publish(PubsubMessage.newBuilder().setData(ByteString.copyFrom("test".getBytes())).build())
-        .get(10, TimeUnit.SECONDS);
+  public void canReadOneMessage() {
+    publish(0, TEST_MESSAGE);
 
     AtomicReference<PubsubMessage> received = new AtomicReference<>();
     AtomicReference<Pubsub.Read> input = new AtomicReference<>();
 
-    input.set(new Pubsub.Read(subscriptionName.toString(),
+    input.set(new Pubsub.Read(subscriptions.get(0).getName(),
```

Comment:
This may be too much magic. It took me a while to realize `subscriptions` was defined in `TestWithPubsubResources`. This sort of thing is normal in Scala, but feels non-idiomatic in Java.

Could we perhaps rename this method to `generatedSubscriptions` or `testSubscriptions`? That might make it a little more obvious that they're coming from the base class? Or maybe we should access it as `TestWithPubsubResources.this.subscriptions`?

Reply:
how about just `getSubscriptions()`?

Reply:
Yes, I like `getSubscriptions` much better.